pkg/controller/controller_instance.go

```diff
@@ -370,81 +389,133 @@ func (c *controller) pollInstance(serviceClass *v1alpha1.ServiceClass, servicePl
 		deleting = true
 	}
 
-	lastOperationRequest := &brokerapi.LastOperationRequest{
-		ServiceID: serviceClass.ExternalID,
-		PlanID:    servicePlan.ExternalID,
+	request := &osb.LastOperationRequest{
+		InstanceID: instance.Spec.ExternalID,
+		ServiceID:  &serviceClass.ExternalID,
+		PlanID:     &servicePlan.ExternalID,
 	}
 	if instance.Status.LastOperation != nil && *instance.Status.LastOperation != "" {
-		lastOperationRequest.Operation = *instance.Status.LastOperation
+		key := osb.OperationKey(*instance.Status.LastOperation)
+		request.OperationKey = &key
 	}
-	resp, rc, err := brokerClient.PollServiceInstance(instance.Spec.ExternalID, lastOperationRequest)
+
+	glog.V(5).Infof("Polling last operation on Instance %v/%v", instance.Namespace, instance.Name)
+
+	response, err := brokerClient.PollLastOperation(request)
 	if err != nil {
-		glog.Warningf("Poll failed for %v/%v  : %s", instance.Namespace, instance.Name, err)
-		return err
-	}
-	glog.V(4).Infof("Poll for %v/%v returned %q : %q", instance.Namespace, instance.Name, resp.State, resp.Description)
+		// If the operation was for delete and we receive a http.StatusGone,
+		// this is considered a success as per the spec, so mark as deleted
+		// and remove any finalizers.
+		if osb.IsGoneError(err) && deleting {
```

Comment:
@MHBauer that shouldn't affect anything for bindings, since they are only synchronous right now.